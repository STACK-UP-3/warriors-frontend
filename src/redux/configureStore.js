import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
