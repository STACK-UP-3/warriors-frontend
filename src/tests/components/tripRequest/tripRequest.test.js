import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import TripRequest from '../../../components/tripRequests/viewTripRequests/tripRequest';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('*************** Testing the TripRequest  component ***************',()=>{
    let tripReq, wrapper;
    const { location } = window;

    beforeEach(()=>{
        delete window.location;
		window.location = { assign: jest.fn(),search:'' };
        
        const historyMock ={ 
            replace: jest.fn(),
            length: 0,
            location: { 
                pathname: '',
                search: '',
                state: '',
                hash: ''
            },
            action: 'REPLACE',
            push: jest.fn(),
            go: jest.fn(),
            goBack: jest.fn(),
            goForward: jest.fn(),
            block: jest.fn(),
            listen: jest.fn(),
            createHref: jest.fn()
        } ;
        
        tripReq = {
            createdAt: '2020-06-12',
            id: 2,
            line_manager_id: 2,
            status: 'pending',
            tripRequest: [{
                origin:'origin city',
                destination:'destination city',
                type: 'one-way-trip',
                travelReason: 'managerTest',
                accommodation_id: 1
                }],
            trip_id: 2,
            updatedAt: '2020-06-12',
            user_id: 1,
        }
        
        wrapper = mount( 
            <Provider store={ store }> 
                <Router history={historyMock}>
                    <TripRequest key={ tripReq.id } information={ tripReq } currentPage={2} /> 
                </Router>
            </Provider>
        );
        
    });
    
    afterAll(() => {
		window.location = location; 
    });

    it('Should render the Trip Request page correctly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
    
    it('Should direct the user to view a specific trip', ()=>{
        const historyMock ={ 
            replace: jest.fn(),
            length: 0,
            location: { 
                pathname: '',
                search: '',
                state: '',
                hash: ''
            },
            action: 'REPLACE',
            push: jest.fn(),
            go: jest.fn(),
            goBack: jest.fn(),
            goForward: jest.fn(),
            block: jest.fn(),
            listen: jest.fn(),
            createHref: jest.fn()
             } ;
        
        const wrapper = mount( 
                <Provider store={ store }> 
                    <Router history={historyMock}>
                        <TripRequest key={ tripReq.id } information={ tripReq } currentPage={2} /> 
                    </Router>
                </Provider>
            );
        
        wrapper.find('[className="collection-item waves-effect waves-tea"]').simulate('click');
        
        expect(historyMock.push).toHaveBeenCalled();
    });
});
