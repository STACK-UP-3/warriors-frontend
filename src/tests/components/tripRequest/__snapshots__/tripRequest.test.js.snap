// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the TripRequest  component *************** Should render the Trip Request page correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 0,
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {
          "hash": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
  >
    <_default
      currentPage={2}
      information={
        Object {
          "createdAt": "2020-06-12",
          "id": 2,
          "line_manager_id": 2,
          "status": "pending",
          "tripRequest": Array [
            Object {
              "accommodation_id": 1,
              "destination": "destination city",
              "origin": "origin city",
              "travelReason": "managerTest",
              "type": "one-way-trip",
            },
          ],
          "trip_id": 2,
          "updatedAt": "2020-06-12",
          "user_id": 1,
        }
      }
      key="2"
    >
      <div
        className="collection"
      >
        <div
          className="collection-item waves-effect waves-tea"
          id={2}
          onClick={[Function]}
        >
          <div
            className="itemDetail"
            id="origin"
          >
            <b>
                 Origin:
            </b>
             
            <span>
               
              origin city
               
            </span>
          </div>
          <div
            className="itemDetail"
            id="destination"
          >
            <b>
               Destination:
            </b>
             
            <span>
               
              destination city
               
            </span>
          </div>
          <div
            className="itemDetail"
            id="manager"
          >
            <b>
               Reason:
            </b>
             
            <span>
               
              managerTest
               
            </span>
          </div>
          <div
            className="itemDetail"
            id="tripType"
          >
            <b>
               Type:
            </b>
             
            <span>
               
              one-way-trip
               
            </span>
          </div>
        </div>
      </div>
    </_default>
  </Router>
</Provider>
`;
